src/Component/FieldTypes/CallableFieldType.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\FieldTypes;
+
+use Sylius\Component\Grid\DataExtractor\DataExtractorInterface;
+use Sylius\Component\Grid\Definition\Field;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class CallableFieldType implements FieldTypeInterface
+{
+    public function __construct(private DataExtractorInterface $dataExtractor)
+    {
+    }
+
+    public function render(Field $field, $data, array $options): string
+    {
+        $value = $this->dataExtractor->get($field, $data);
+        $value = call_user_func($options['callable'], $value);
+
+        try {
+            $value = (string) $value;
+        } catch (\Throwable) {
+            throw new \RuntimeException(\sprintf('The callback for field "%s" returned a value that could not be converted to string.', $field->getName()));
```

Comment:
How about this?

Symfony uses this new syntax in the docs quite a lot
> Object (id 1) is not found.


```suggestion
            throw new \RuntimeException(\sprintf('Callable field (name "%s") returned value could not be converted to string.', $field->getName()));
```

Additionally we can either show actual error via `\Throwable $e->getMessage()` and get type of returned value via `get_debug_type()`